drivers/ethernet/eth_stm32_hal.c

```diff
@@ -326,16 +323,21 @@ void HAL_ETH_TxPtpCallback(uint32_t *buff, ETH_TimeStampTypeDef *timestamp)
 }
 #endif /* CONFIG_PTP_CLOCK_STM32_HAL */
 
-#if defined(CONFIG_ETH_STM32_HAL_API_V2)
 static int eth_tx(const struct device *dev, struct net_pkt *pkt)
 {
 	struct eth_stm32_hal_dev_data *dev_data = dev->data;
 	ETH_HandleTypeDef *heth = &dev_data->heth;
 	int res;
 	size_t total_len;
+#if defined(CONFIG_ETH_STM32_HAL_API_V2)
 	size_t remaining_read;
 	struct eth_stm32_tx_context *ctx = NULL;
 	struct eth_stm32_tx_buffer_header *buf_header = NULL;
+#else
+	uint8_t *dma_buffer;
+	__IO ETH_DMADescTypeDef *dma_tx_desc;
+
```

Comment:
@erwango, I think you should drop the extra empty line below and straight revert the targeted commit.
CI "Run compliance checks" is wrong here, it's not clear why.